crcmocks/rbac.py

```diff
@@ -1,33 +1,83 @@
 
-rbac_response = {
-    "meta": {"count": 30, "limit": 1000, "offset": 0},
+BASE_RBAC_RESPONSE = {
+    "meta": {"count": None, "limit": None, "offset": 0},
     "links": {
         "first": "/api/rbac/v1/access/?application=&format=json&limit=1000&offset=0",
         "next": None,
         "previous": None,
         "last": "/api/rbac/v1/access/?application=&format=json&limit=1000&offset=0",
     },
-    "data": [
-        {"resourceDefinitions": [], "permission": "advisor:*:*"},
-        {"resourceDefinitions": [], "permission": "inventory:*:*"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:*"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-        {"resourceDefinitions": [], "permission": "inventory:*:read"},
-    ],
+    "data": [],
 }
 
 
-# TODO: allow customizable rbac response per user
-@blueprint.route("/v1/access/")
+# TODO: support resource definitions?
+@blueprint.route("/v1/access/", methods=["GET"])
 def rbac_access():
+    application = request.args.get("application")
+    limit = int(request.args.get("limit", 100))
+    offset = int(request.args.get("offset", 0))
+    # get the identity header
+    identity_header = request.headers.get("X-Rh-Identity")
+
+    if not identity_header:
+        return "No identity header present in request\n", 401
+
+    # get the user
+    user, username, account_number = get_user_rh_identity(identity_header)
+
+    if not user:
+        return f"No user found in TinyDB with username:" \
+               f" {username} or account_number: {account_number}\n", 404
+    if len(user) > 1:
+        return f"Multiple users found with:" \
+               f" {username} or account_number: {account_number}\n", 406
+
+    # get the user permissions
+    if user[0].get("entitlements") is not None:
+        permissions = user[0]["permissions"].split(",")
+    else:
+        permissions = DEFAULT_PERMISSIONS.copy()
+
+    rbac_response = deepcopy(BASE_RBAC_RESPONSE)
+    # set links
+    rbac_response["links"]["first"] = f"/api/rbac/v1/access/?application={application or ''}" \
+                                      f"&format=json&limit={limit}" \
+                                      f"&offset={offset}"
+    rbac_response["links"]["last"] = f"/api/rbac/v1/access/?application={application or ''}" \
+                                     f"&format=json&limit={limit}" \
+                                     f"&offset={offset}"
+
+    limit_count = 1
+    # TODO: implement 'real' pagination
+    for i, perm in enumerate(permissions):
+        if i >= offset:
+            if application:
```

Comment:
This block confuses me, seems we can just always add the perm to the rbac_response? Since the one you're adding after this if there is an application doesn't split, but just assigns the same perm. Should we even care at this point if there is an application?

Reply:
If we're filtering by a comma-separated list of applications, we only want to return the permission for that specific application(s). 

`perm.split(":")[0]` gets the application of the user's permission and checks if it is in the application string that was attached to the request. 

Otherwise we'd just be returning all the user's permissions and disregarding the `application` filter.

Reply:
Oh I see, if we have an application object this would always occur. I'll chalk this up to a brain fart 😂